crates/ruff_linter/src/settings/fix_safety_table.rs

```diff
@@ -0,0 +1,166 @@
+use std::fmt::Debug;
+
+use ruff_macros::CacheKey;
+use rustc_hash::FxHashMap;
+use strum::IntoEnumIterator;
+
+use crate::{
+    registry::{Rule, RuleSet},
+    rule_selector::{PreviewOptions, Specificity},
+    RuleSelector,
+};
+
+/// A table to keep track of which rules fixes should have
+/// their safety overridden.
+#[derive(Debug, CacheKey, Default)]
+pub struct FixSafetyTable {
+    forced_safe: RuleSet,
+    forced_unsafe: RuleSet,
+}
+
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum FixSafety {
+    ForcedSafe,
+    ForcedUnsafe,
+    Default,
+}
```

Comment:
I'd rather not add another enum, I think this may be removable with my suggestion to provide a `resolve_applicability` interface?